tests/integration/test_issue_2006.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import os.path
+import subprocess
+
+from pex.testing import run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+def test_packaging(tmpdir):
+    # type: (Any) -> None
+
+    pex = os.path.join(str(tmpdir), "speak.pex")
+    run_pex_command(
+        args=["conscript==0.1.5", "cowsay==5.0", "fortune==1.1.0", "-c", "conscript", "-o", pex]
+    ).assert_success()
+
+    assert (
+        "5.0" == subprocess.check_output(args=[pex, "cowsay", "--version"]).decode("utf-8").strip()
+    )
+
+    cowsay = os.path.join(str(tmpdir), "cowsay")
+    os.symlink(pex, cowsay)
+    assert "5.0" == subprocess.check_output(args=[cowsay, "--version"]).decode("utf-8").strip()
```

Comment:
Probably worth adding a similar comment in the test explaining this. It's not obvious to the reader why this test would fail without the fix, and the answer is subtle (requires knowing something about how conscript uses symlink names)

Reply:
Agreed, coming up. That said, I'll let the conscript case be the only test with the comment accounting for the context needed.